homeassistant/components/opensprinkler/binary_sensor.py

```diff
@@ -0,0 +1,128 @@
+"""Opensprinkler integration."""
+import logging
+from typing import Callable
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.core import HomeAssistant
+
+from . import OpensprinklerBinarySensor
+from .const import DATA_DEVICES, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: dict,
+    async_add_entities: Callable,
+    discovery_info: dict,
+):
+    """Set up the opensprinkler binary sensors."""
+    entities = await hass.async_add_executor_job(
+        _create_entities, hass, config, discovery_info
+    )
+    async_add_entities(entities)
+
+
+def _create_entities(hass: HomeAssistant, config: dict, discovery_info: dict):
+    entities = []
+
+    name = discovery_info["name"]
+    device = hass.data[DOMAIN][DATA_DEVICES][name]
+    entities.append(DeviceBinarySensor(name, device, None, "operation_enabled"))
+    entities.append(
+        DeviceBinarySensor(f"{name} Rain Delay", device, None, "rain_delay")
+    )
+
+    fwv = device.device.firmware_version
+    hwv = device.device.hardware_version
+    if fwv >= 219:
+        entities.append(
+            DeviceBinarySensor(
+                f"{name} Rain Sensor 1", device, "moisture", "rain_sensor_1",
+            )
+        )
+        if hwv / 30 >= 1:
```

Comment:
You can write this as
```suggestion
        if hwv >= 30:
```